### PR TITLE
OC-749: Displaying additional information fields on publications

### PR DESCRIPTION
**Pull request by finlay-jisc**
The purpose of this PR was to display the new additional information fields when viewing a publication webpage and a publication PDF.

---

### Acceptance Criteria:

- On the publication page, immediately below the main text (above the sections for linked publications), a heading is present for “Additional parts of this work hosted elsewhere” with a list of each, displayed with:
    - The title in bold
    - The description immediately on the line below the title
    - The URL for the resource displayed as a hyperlink
        - This link should be marked up using the link’s title in the format “Link to <title>”, to be announced to screen readers
- On PDFs, a section is present to display additional information, including the section title “Additional parts of this work hosted elsewhere” in the same font size/style as other heading styles such as “Funders”, “Conflicts of Interest”, “Parent Publications” etc.
    - Below the section heading, each additional info link is displayed with it’s titled in bold, followed by it’s description in body-style text, and a hyperlink of its URL.
        - If possible, this hyperlink is marked up to read the title to screen readers
        - A double linebreak (or similar spacing) is present between each additional info link to make it easier for readers to distinguish them.

---

### Checklist:

- [x] Local manual testing conducted
- [x] Automated tests added
- [ ] Documentation updated

---

### Tests:

UI
Screenshot 2024-01-25 135304

API
Screenshot 2024-01-25 114748

E2E
Screenshot 2024-01-25 135243

---

### Screenshots:

Publication page
Screenshot 2024-01-25 144821

PDF
Screenshot 2024-01-25 144921

### CODE REVIEW

**Review by adwearing-jisc — DISMISSED** [review]

File: ui/src/components/AdditionalInformationCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+type Props = {
+    additionalInformation: {
+        title: string;
+        url: string;
+        description?: string;
+    };
+};
+
+const AdditionalInformationCard: React.FC<Props> = (props): React.ReactElement => (
```
Comment (adwearing-jisc) [inline ui/src/components/AdditionalInformationCard/index.tsx:11]: Confused what React.ReactElement is doing here. Could also destructure the props

`const AdditionalInfoCard: React.FC<Props> = ({ additionalInformation: { title, url, description }}) => {`

Comment (finlay-jisc) [reply ui/src/components/AdditionalInformationCard/index.tsx]: The React.ReactElement is just to specify the return type of the function component for typescript. I've destructured the props now too

File: ui/src/pages/publications/[id]/versions/[versionId].tsx [hunk 2]
```diff
@@ -215,6 +215,7 @@ const Publication: Types.NextPage<Props> = (props): React.ReactElement => {
 
     const list = [];
 
+    const showAdditionalInformation = Boolean(publicationVersion?.additionalInformation.length);
```
Comment (adwearing-jisc) [inline ui/src/pages/publications/[id]/versions/[versionId].tsx:218]: Curious way to cast to a boolean (I know you're keeping things standard with what's already there).. common way I use is `!!publicationVersion?.additionalInformation.length`

**Review by adwearing-jisc — APPROVED** [review]

File: ui/src/pages/publications/[id]/versions/[versionId].tsx (hunk 2 above)
Comment (finlay-jisc) [reply ui/src/pages/publications/[id]/versions/[versionId].tsx:218]: It does read a bit old fashioned. I'd probably use `!!` too in most cases. I do like how explicit this way is at least, even if it is longer

**Pull request merged**